curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -1012,6 +1012,24 @@ void ChunkCacheManager::ReleaseCache(S3ClientAdaptorImpl* s3ClientAdaptor) {
     dataRCacheMap_.clear();
 }
 
+void ChunkCacheManager::ReleaseWriteDataCache(uint64_t key,
+     uint64_t dataCacheLen) {
+    WriteLockGuard writeLockGuard(rwLockWrite_);
+    if (dataWCacheMap_.count(key)) {
+        dataWCacheMap_.erase(key);
+        s3ClientAdaptor_->GetFsCacheManager()->DataCacheNumFetchSub(1);
+        LOG(INFO) << "chunk flush DataCacheByteDec1 len:"
+                  << dataCacheLen;
+        s3ClientAdaptor_->GetFsCacheManager()->
+            DataCacheByteDec(dataCacheLen);
+        if (!s3ClientAdaptor_->GetFsCacheManager()
+            ->WriteCacheIsFull()) {
+            LOG(INFO) << "write cache is not full, signal wait.";
+            s3ClientAdaptor_->GetFsCacheManager()->FlushSignal();
```

Comment:
it seems these code are not necessary protected by lock

Reply:
fixed